### PR TITLE
[KYUUBI #5757][AUTHZ] Support alter path-based table for Delta Lake in Authz

### LINKED ISSUE #5757: [Subtask] Support alter path-based table for Delta Lake in Authz
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Support alter path-based table for Delta Lake in Authz plugin.

### Parent issue

#5663

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zml1206**
# :mag: Description
## Issue References 🔗

This pull request fixes #5757 

## Describe Your Solution 🔧

Add uriDescs.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [x] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -468,6 +477,95 @@ class DeltaCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+    })
+  }
+
+  test("alter path-based table add columns") {
+    withTempDir(path => {
+      doAs(admin, sql(createPathBasedTableSql(path)))
+      val addColumnsSql = s"ALTER TABLE delta.`$path` ADD COLUMNS (age int)"
+      interceptEndsWith[AccessControlException](
+        doAs(someone, sql(addColumnsSql)))(
+        s"does not have [write] privilege on [[$path, $path/]]")
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:499]: this behavior looks strange to me, add a column seems to have nothing to do with the path..

Comment (zml1206) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:499]: This is how I understand it. The permissions of a path-based table are only read and write. alter add column modifies the schema of the table, so write permission is required.

**Comment by yaooqinn** [comment]
Thanks, merged to master

**Comment by yaooqinn** [comment]
Sorry, reverted and reopened it via conflicts

**Comment by zml1206** [comment]
Conflit resolved, @yaooqinn

**Comment by zml1206** [comment]
> > Sorry, reverted and reopened it via conflicts
> 
> Thanks for your contribution. @zml1206
> 
> Because the master's CI fails now, I will revert this change first.
> 
> [b920836](https://github.com/apache/kyuubi/commit/b920836e03cf20d2e1845d890cc3a8404037b3ec)

Already resubmitted. @cxzl25 cc @yaooqinn

**Comment by zml1206** [comment]
Thanks, @yaooqinn @cxzl25

**Pull request merged**